modules/administration/pages/maintenance-window.adoc

```diff
@@ -1,33 +1,39 @@
 [[maintenance-window]]
-= Maintenance Window Checklist
+= {productname} Maintenance Window
 
-If you work on a {productname} Server with scheduled maintenance windows, you might find it difficult to remember all the things that you need to during that critical downtime.
+If you work with scheduled maintenance windows, you might find it difficult to remember all the things that you need to do before, during, and after that critical downtime of the {productname} Server.
+{productname} Server related systems such as Inter-Server Synchronization Slave Servers or {productname} Proxies are also affected and have do be considered.
+
+{suse} recommends to keep the {productname} infrastructure (servers, proxies, build hosts, etc.) always up-to-date.
+Because the {productname} Server is needed to update all your IT landscape, the server must always be updated.
+// It's similar to zypper at the package level:
+You must keep it up-to-date or you will risk not being able to update something else successfully!
 
-As with any {sle} system, you are encouraged to install package updates on the {susemgr} Server when they arrive.
 
-This section contains a checklist for your maintenance window, with links to further information on performing each of the steps, if you need it.
 
+== {productname} Server
+
+This section contains a checklist for your maintenance window, with links to further information on performing each of the steps, if you need it.
 
-. Stop services.
-You will need to stop the spacewalk, SAP, and database services, along with any others you have running.
+// ke, 2019-09-30: we'll stop spacewalk during the update
+// . Stop spacewalk services.
+// You will need to stop the spacewalk, SAP, and database services, along with any others you have running.
 . Check if the configuration is still correct.
-. On Salt systems, apply the highstate.
 . Apply the latest updates.
 See xref:upgrade:server-update.adoc[].
 . Update to the latest service pack, if required.
 See xref:upgrade:migrate-4x-4x.adoc[].
```

Comment:
This link points to a document with the following introduction:
 > Migrating {productname} from version 4 to 4.x (when available) can be done using either the {yast} Online Migration tool, or with the Zypper command line tool.

We should maybe mention here:

> Update {productname} Server to the latest major version available, if required
> See xref:upgrade:migrate-4x-4x.adoc[].

Reply:
I guess both kinds of migrations should actually be mentioned: SP upgrades of the underlying SLES/openSUSE minor version and minor product upgrades.

Reply:
I see your point and I'm mostly for it.  But because SUSE Manger now is are product (and not just an Extension), I'm wondering whether we still want to talk that much about the underlying OS.  It probably makes sense if "zypper migration" will still offer such migration when available:
```
SLES 15 SP1 with SUMA 4.0
  |
  +-> SLES 15 SP2 with SUMA 4.0
```

Reply:
@juliogonzalez 
> We should maybe mention here:
> 
> > Update {productname} Server to the latest major version available, if required
> > See xref:upgrade:migrate-4x-4x.adoc[].

Hhmm, no or I'm not sure what you mean :) The upgrade scenario ("migration") usually does not work or is not recommended for major version upgrades -- for this, you need the offline procedure with autoyast (or simply booting from a boot medium and then run the update/upgrade). This is what we have in migrate-3x-4x.adoc (IMO).

If things become available we should describe them as needed.  SUMA 5 on SLES 15.x, maybe. SLES 16 running SUMA 4.x, probably not, but who knows...

The main problem usually is that nobody informs us early enough, and nobody paints the complete picture.  Of course, nobody knows the complete picture. And those who know a lot of it, often cannot imagine how stupid we are ;)

Reply:
Yeah, "latest major version available" is not the correct sentence. It should be "latest Maintenance Update available for your current major version`.

Let me explain it more clearly.

Talking about SUSE Manager, we basically use a `X.Y.Z` versioning.

`Z` changes: that means a MU. User stops services, updates packages, runs the script to update the database, starts services, end of the story: Example 4.0.1 -> 4.0.2

`Y` changes: Means that a new major version, but that allows an in-line "migration". You don't replace the server. It usually means you will jump to a new SLE service pack. Bue before starting you need a fully updated available `Z` version of your `Y`. Example: 3.1.11 to 3.2.0.

`X` changes: Means that's a breaking change, that doesn't allow an in-line "migration". You need your current server updated to the last `Y.Z` of your `X` version. And you need a new server with the new version, so you can migrate your data. Example: 3.2.11 to 4.0.0.

So FMPOV that `xref:upgrade:migrate-4x-4x.adoc[]` is like a document for the future, describing what happens when `Y` changes. That is: when we will have `4.1.0` available, so users can migrate from `4.0.X`, as it says:

> Migrating {productname} from version 4 to 4.x (when available) can be done using

And where you say `4' , I guess it should say `4.0`.

In this case, the users will need to be at the latest MU available for `4.0` before being able to migrate to `4.1`

Right now we need to have guides for "Z changes" (MUs), "X changes" (migration from 3.2 to 4.0), and you can already prepare "Y changes" (4.0 -> 4.1), but I think we should not offer it to clients yet.

Reply:
"Z changes" (MUs) are accumulated and consolidated package updates, IMO.  And this is what I want to see covered in the server-update.adoc file.  I basically agree with all you said here :)

Reply:
Yes @keichwa.

There's two extra twists to keep in mind: underlying OS SP migrations and Postgres migrations are typically done in between "Y changes".

An example progression from the past versions could be:

 - 3.1.4, SLES 12 SP3, Postgres 9.6
 - 3.1.11, SLES 12 SP3, Postgres 9.6
 - 3.2.10, SLES 12 SP3, Postgres 9.6
 - 3.2.10, SLES 12 SP3, Postgres 10
 - 3.2.10, SLES 12 SP4, Postgres 10
 - 4.0.1, SLES 15 SP1, Postgres 10

Package upates only cover "Z changes" (MUs), SP migrations and "Y changes" need `zypper migrate`, Postgres needs the old and new versions to be installed in parallel plus a manual procedure to migrate data, and "X changes" require a new server altogether.

Hopefully all cases are covered above!

Reply:
I added a remark about the database schema upgrade.

Ths version info went into the db-migration file. For the moment, it's commented.